src/test/java/games/strategy/engine/framework/ArgParserTest.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.engine.framework;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+
+import org.junit.Test;
+
+public class ArgParserTest {
+
+
+  @Test
+  public void argsTurnIntoSystemProps() {
+    assertThat("check precondition, system property for our test key should not be set yet.",
+        System.getProperty(TestData.propKey), nullValue());
+
+    boolean result = ArgParser.handleCommandLineArgs(
+        TestData.sampleArgInput, TestData.samplePropertyNameSet);
+
+    assertThat("prop key was supplied as an available value, "
+            + " which was passed as a test value - everything should "
+            + "have parsed well, expect true result",
+        result, is(true));
+    assertThat("system property should now be set to our test value",
+        System.getProperty(TestData.propKey), is(TestData.propValue));
```

Comment:
No worries, @DanVanAtta.  Was just pointing this out as I have been burned in the past by shared state across tests on more than one occasion.  (It's more likely to be a problem in a different fixture whose behavior is transitively dependent on one of these system properties, which I think your `@After` block addresses)  I agree with your statement about asserting pre-conditions used to detect this problem, which will clearly identify the cause of failure without having to do much digging.